examples/remote-sensing/plot_clustering.py

```diff
@@ -0,0 +1,373 @@
+    if not os.path.exists(data_path):
+        os.makedirs(data_path, exist_ok=True)
+    if not all([os.path.exists(os.path.join(data_path, filename))
+                for filename in filenames]):
+        print('Downloading Salinas dataset...')
+        for url in urls:
+            urllib.request.urlretrieve(url, os.path.join(data_path,
+                                                 os.path.basename(url)))
+        print('Done.')
+    else:
+        print('Salinas dataset already downloaded.')
+
+
+def read_salinas(data_path: str, version: str="corrected") -> \
+        Tuple[ArrayLike, ArrayLike, Dict[int, str]]:
+    """
+    Read Salinas hyperspectral data.
+
+    Parameters
+    ----------
+    data_path: str
+        Path to the data folder
+    version: str
+        Version of the data to read. Can be either "corrected" or "raw".
+        Default is "corrected".
+
+
+    Returns
+    -------
+    data: ArrayLike
+        Data array of shape (512, 217, 204).
+
+    labels: ArrayLike
+        Labels array of shape (512, 217).
+
+    labels_names: Dict[int, str]
+        Dictionary mapping labels to their names.
+    """
+    if version == "corrected":
+        data_file = os.path.join(data_path, "Salinas_corrected.mat")
+    else:
+        data_file = os.path.join(data_path, "Salinas.mat")
+    data = loadmat(data_file)['salinas_corrected']
+    labels = loadmat(os.path.join(data_path,
+                                  'Salinas_gt.mat'))['salinas_gt']
+    labels_names = {
+                0: 'Undefined',
+                1: 'Brocoli_green_weeds_1',
+                2: 'Brocoli_green_weeds_2',
+                3: 'Fallow',
+                4: 'Fallow_rough_plow',
+                5: 'Fallow_smooth',
+                6: 'Stubble',
+                7: 'Celery',
+                8: 'Grapes_untrained',
+                9: 'Soil_vinyard_develop',
+                10: 'Corn_senesced_green_weeds',
+                11: 'Lettuce_romaine_4wk',
+                12: 'Lettuce_romaine_5wk',
+                13: 'Lettuce_romaine_6wk',
+                14: 'Lettuce_romaine_7wk',
+                15: 'Vinyard_untrained',
+                16: 'Vinyard_vertical_trellis'
+            }
+    return data, labels, labels_names
+
+
+def dowload_uavsar(data_path: str):
+    """Download the UAVSAR dataset.
+    Parameters
+    ----------
+    data_path: str
+        Path to the data folder to download the data.
+    """
+    url = 'https://zenodo.org/records/10625505/files/scene1.npy?download=1' 
+    filename = "scene1.npy"
+    if not os.path.exists(data_path):
+        os.makedirs(data_path, exist_ok=True)
+    if not os.path.exists(os.path.join(data_path, filename)):
+        print('Downloading UAVSAR dataset...')
+        urllib.request.urlretrieve(url, os.path.join(data_path, filename))
+        print('Done.')
+    else:
+        print('UAVSAR dataset already downloaded.')
+
+
+if __name__ == "__main__":
```

Comment:
`if __name__ == "__main__"` is not necessary. You can use a #-line to separate class definition from the actual example execution.
```suggestion
###############################################################################
```
Don't forget to remove indent for the code below.